eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/ConcurrentHashMap.java

```diff
@@ -1979,6 +1979,12 @@ public void zeroOutQueuePosition()
         return (ConcurrentHashMap<K, V>) super.withKeyValue(key, value);
     }
 
+    @Override
+    public ConcurrentHashMap<K, V> withMap(Map<? extends  K, ? extends V> map)
+    {
+        return (ConcurrentHashMap<K, V>) super.withMap(map);
```

Comment:
Any reason not to do this.putAll(); return this; ?

Reply:
I basically followed the same coding style here for all `with*` methods! only reason is consistency

Reply:
@molin Can you take a look at this and review/approve? I took a quick look and didn't see any casts.